src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/ReadWriteSplittingPlugin.java

```diff
@@ -115,27 +123,46 @@ public Object execute(
     if (METHOD_SET_READ_ONLY.equals(methodName) && args != null && args.length > 0) {
       switchConnectionIfRequired((Boolean) args[0]);
     }
+
     try {
-      return this.nextPlugin.execute(methodInvokeOn, methodName, executeSqlFunc, args);
+      final Object result = this.nextPlugin.execute(methodInvokeOn, methodName, executeSqlFunc, args);
+
+      if (this.explicitlyReadOnly && this.loadBalanceReadOnlyTraffic && isTransactionBoundary(methodName)) {
```

Comment:
It's a bit unusual to change connection after execute(), not before as I'd expect.

Reply:
Good point. I thought of one potentially strange scenario that might occur with that change: with autocommit on, calling setReadOnly(true) followed by execute. After calling setReadOnly, the driver would switch to a reader connection. Then, when execute is called, the driver would switch to another reader connection before executing the statement. Is it problematic that it is abandoning the first reader connection without doing anything meaningful in this scenario?